OpenRA.Mods.Common/Traits/World/Selection.cs

```diff
@@ -15,17 +15,24 @@
 using OpenRA.Primitives;
 using OpenRA.Traits;
 
-namespace OpenRA
+namespace OpenRA.Mods.Common.Traits
 {
-	public class Selection
+	public class SelectionInfo : ITraitInfo
+	{
+		public object Create(ActorInitializer init) { return new Selection(this); }
+	}
+
+	public class Selection : ISelection
 	{
 		public int Hash { get; private set; }
 		public IEnumerable<Actor> Actors { get { return actors; } }
 
 		readonly HashSet<Actor> actors = new HashSet<Actor>();
-		readonly INotifySelection[] worldNotifySelection;
+		INotifySelection[] worldNotifySelection = new INotifySelection[0];
+
+		public Selection(SelectionInfo info) { }
 
-		internal Selection(IEnumerable<INotifySelection> worldNotifySelection)
+		public void SetNotifySelection(IEnumerable<INotifySelection> worldNotifySelection)
```

Comment:
This can change to `INotifyCreated.Created`, querying `WorldActor.TraitsImplementing<INotifySelection>()` instead of relying on the World actor to pass the traits.  This can then be removed from the `ISelection` interface.

Reply:
Then I do this way, I have NullPointer Exception. world.WorldActor is null  at this point.
Let me know how I can get WorldActor here?

Reply:
`self`